Sources/Disposable.swift

```diff
@@ -83,10 +84,9 @@ public final class AnyDisposable: Disposable {
 		}
 
 		func dispose() {
-			if state.tryDispose() {
-				action?()
-				action = nil
-			}
+			guard state.tryDispose() else { return }
+			action?()
+			action = nil
```

Comment:
I prefer `if` over `guard` here. I think it communicates the intent more clearly.

Reply:
Hmm, disagreed.
By returning asap, the intention of do nothing if already disposed would be clarified.

Reply:
Thinking about it a bit more, I dislike `guard` here because `state.tryDispose()` is effectful. `guard` strongly suggests a test that's free of effects IMO. I still think it should stay as `if` here and below.

Reply:
Got it 👍